pkg/controller/perconaservermongodb/pbm.go

```diff
@@ -0,0 +1,503 @@
+		return nil
+	}
+
+	mainStgName, mainStg, err := cr.Spec.Backup.MainStorage()
+	if err != nil {
+		// no storage found
+		return nil
+	}
+
+	main, err := backup.GetPBMConfig(ctx, r.client, cr, mainStg)
+	if err != nil {
+		return errors.Wrap(err, "get config")
+	}
+
+	// PITR config will be handled by reconcilePiTRConfig function
+	main.PITR = nil
+
+	var external []config.Config
+	for name, stg := range cr.Spec.Backup.Storages {
+		if name == mainStgName {
+			continue
+		}
+
+		profile, err := backup.GetPBMProfile(ctx, r.client, cr, name, stg)
+		if err != nil {
+			return errors.Wrap(err, "get profile")
+		}
+
+		external = append(external, profile)
+	}
+
+	c := append([]config.Config{main}, external...)
+	slices.SortFunc(c, func(a config.Config, b config.Config) int {
+		if a.IsProfile && !b.IsProfile {
+			return 1
+		}
+
+		if !a.IsProfile && b.IsProfile {
+			return -1
+		}
+
+		return strings.Compare(a.Name, b.Name)
+
+	})
+
+	hash, err := hashPBMConfiguration(c)
+	if err != nil {
+		return errors.Wrap(err, "hash config")
+	}
+
+	if cr.Status.BackupConfigHash == hash {
+		return nil
+	}
+
+	log.Info("configuration changed", "oldHash", cr.Status.BackupConfigHash, "newHash", hash)
+
+	pbm, err := backup.NewPBM(ctx, r.client, cr)
+	if err != nil {
+		return errors.Wrap(err, "new PBM connection")
+	}
+
+	currentCfg, err := pbm.GetConfig(ctx)
+	if err != nil && !backup.IsErrNoDocuments(err) {
+		return errors.Wrap(err, "get current config")
+	}
+
+	if currentCfg == nil {
+		currentCfg = new(config.Config)
+	}
+
+	if err := pbm.GetNSetConfig(ctx, r.client, cr); err != nil {
+		return errors.Wrap(err, "set config")
+	}
+
+	if !reflect.DeepEqual(currentCfg.Storage, main.Storage) {
+		log.Info("resync storage", "storage", mainStgName)
+
+		if err := pbm.ResyncStorage(ctx, &main.Storage); err != nil {
+			return errors.Wrap(err, "resync")
+		}
+	}
+
+	cr.Status.BackupConfigHash = hash
+
+	return nil
+}
+
+func hashPBMConfiguration(c []config.Config) (string, error) {
+	v, err := json.Marshal(c)
+	if err != nil {
+		return "", err
+	}
+
+	return sha256Hash(v), nil
+}
+
+func (r *ReconcilePerconaServerMongoDB) reconcilePiTRConfig(ctx context.Context, cr *psmdbv1.PerconaServerMongoDB) error {
+	log := logf.FromContext(ctx)
+
+	if !cr.Spec.Backup.Enabled {
+		return nil
+	}
+
+	if cr.PBMResyncNeeded() || cr.PBMResyncInProgress() {
+		return nil
+	}
+
+	// pitr is disabled right before restore so it must not be re-enabled during restore
+	isRestoring, err := r.isRestoreRunning(ctx, cr)
+	if err != nil {
+		return errors.Wrap(err, "checking if restore running on pbm update")
+	}
+
+	if isRestoring || cr.Status.State != psmdbv1.AppStateReady {
+		return nil
+	}
+
+	stgName, _, err := cr.Spec.Backup.MainStorage()
+	if err != nil {
+		// no storage found
+		return nil
+	}
+
+	pbm, err := backup.NewPBM(ctx, r.client, cr)
+	if err != nil {
+		return errors.Wrap(err, "create pbm object")
+	}
+	defer pbm.Close(ctx)
```

Comment:
Since `pbm` is used on this scope for the first, here

```
	if err := enablePiTRIfNeeded(ctx, pbm, cr); err != nil {
		return errors.Wrap(err, "enable pitr if needed")
	}

```

We can save some API calls if we follow full backup logic returns by moving the new pbm creation after the if clause.